compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingInterfaces.td

```diff
@@ -264,4 +264,80 @@ def IREEEncoding_EncodingTypeInterface :
   ];
 }
 
+//===----------------------------------------------------------------------===//
+// Encoding Propagation Attribute & Op Interfaces
+//===----------------------------------------------------------------------===//
+
+def IREEEncoding_PropagationAttrInterface :
+  AttrInterface<"EncodingPropagationAttrInterface"> {
+  let cppNamespace = "::mlir::iree_compiler::IREE::Encoding";
+  let description = [{
+    Interface used to query new encoding attributes that can be propagated to
+    the operands and result of the operation.
+  }];
+
+  let methods = [
+    InterfaceMethod<
+        [{
+          Returns true if the encodings can be propagated across the operation.
+        }],
+        /*retTy=*/"bool",
+        /*methodName=*/"isPropagable",
+        /*args=*/(ins
+          "::mlir::Value":$target
+        ),
+        /*methodBody=*/"",
+        /*defaultImplementation=*/[{
+            return false;
+        }]
+    >,
+    InterfaceMethod<
+        [{
+          Returns the new encodings for operand and result types
+          for the given operation.
+        }],
+        /*retTy=*/
+        "llvm::FailureOr<::mlir::iree_compiler::IREE::Encoding::PropagationEncoding>",
+        /*methodName=*/"generateEncodings",
+        /*args=*/(ins
+          "::mlir::Value":$target
+        ),
+        /*methodBody=*/"",
+        /*defaultImplementation=*/[{
+          return failure();
+      }]
+    >
+  ];
+}
+
+def IREEEncoding_PropagationOpInterface :
+  OpInterface<"EncodingPropagationOpInterface"> {
+  let cppNamespace = "::mlir::iree_compiler::IREE::Encoding";
+  let description = [{
+      Interface used to generate new operations with new encodings.
+  }];
+
+  let methods = [
+    InterfaceMethod<
+        [{
+          Transforms the op with given encodings. If succeeded, it replaces
+          the uses of the op except the opResult.
```

Comment:
> If succeeded, it replaces the uses of the op except the opResult.

Is this statement correct? I don't see the code that reflects it, can you point it out or update the description?